compiler/rustc_mir_build/src/thir/constant.rs

```diff
@@ -58,8 +53,21 @@ pub(crate) fn lit_to_const<'tcx>(
                 tcx.sess.delay_span_bug(DUMMY_SP, "encountered LitKind::Err during mir build"),
             ));
         }
+        (_, ty::Projection(ty::ProjectionTy { substs, item_def_id })) => {
+            let binder_ty = tcx.bound_type_of(*item_def_id);
+            let ty = binder_ty.subst(tcx, substs);
+            get_valtree(tcx, ty, neg, lit)?
+        }
         _ => return Err(LitToConstError::TypeError),
-    };
+    })
+}
+
+pub(crate) fn lit_to_const<'tcx>(
+    tcx: TyCtxt<'tcx>,
+    lit_input: LitToConstInput<'tcx>,
+) -> Result<ty::Const<'tcx>, LitToConstError> {
+    let LitToConstInput { lit, ty, neg } = lit_input;
```

Comment:
wherever that type is coming from, that's where it should have been normalized imo.

Reply:
Done!